source/common/http/filter_manager.cc

```diff
@@ -860,6 +860,10 @@ void FilterManager::sendLocalReply(
   }
 
   stream_info_.setResponseCodeDetails(details);
+  if (grpc_status.has_value()) {
```

Comment:
I'm likely missing something here, but why not add the status into the headers/trailers at this point, rather than adding it to stream into and then pulling it back out to add it to headers?

Reply:
As I mentioned in this PR description, it won't work for the case for HTTP downstream, gRPC upstream and using gRPC transcoder in envoy, where gRPC status cannot be passed in headers or trailers.

Reply:
So the problem is you're setting the grpc_status in your sendLocalReply, it's being dropped because Envoy is sending a non-gRPC reply, but you want the local reply's grpc status in your access logs? 
I'm having a bit of a hard time justifying adding stream info size for all Envoy users to capture gRPC status for non-gRPC responses.  

Given that local replies will traverse your response filters, how about you have a filter which implements onLocalReply(const LocalReplyData&)   That'll let you capture the grpc_status which is sent in the local reply, insert it using the extensible fields in stream info.

alternately while Envoy isn't going to add a grpc-status for non-grpc responses there's nothing from stopping you from doing so which I believe would also solve the problem.

cc @snowp @lizan for second thoughts.
/wait

Reply:
In fact, gRPC status isn't in [LocalReplyData](https://github.com/envoyproxy/envoy/blob/main/envoy/http/filter.h#L685-L693) used in `onLocalReply(const LocalReplyData&)`. I guess you mean this [one](https://github.com/envoyproxy/envoy/blob/main/source/common/http/utility.h#L369).

I am fine with this approach and I can add gRPC status into [LocalReplyData](https://github.com/envoyproxy/envoy/blob/main/envoy/http/filter.h#L685-L693)  in this PR.

Reply:
> I'm having a bit of a hard time justifying adding stream info size for all Envoy users to capture gRPC status for non-gRPC responses.

+1, I'm not clear either why do you want to capture this.

> alternately while Envoy isn't going to add a grpc-status for non-grpc responses there's nothing from stopping you from doing so which I believe would also solve the problem.

This also works, note that you can even emit trailers to HTTP/1.1 request and the codec will disregard them by default.

Reply:
Thanks guys! Can you please take another look?